src/modeler/AutoRest.Swagger/Validation/BodyTopLevelProperties.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+using AutoRest.Swagger.Model;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class BodyTopLevelProperties : TypedRule<Dictionary<string, Operation>>
+    {
+
+        private readonly Regex resourceRefRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
+        private readonly string[] allowedTopLevelProperties = { "name", "type", "id", "location", "properties", "tags", "plan", "sku", "etag",
+                                                                "managedBy", "identity"}; 
+        /// <summary>
+        /// This rule passes if the body parameter contains top level properties only from the allowed set: name, type,
+        /// id, location, properties, tags, plan, sku, etag, managedBy, identity
+        /// </summary>
+        /// <param name="paths"></param>
+        /// <returns></returns>
+        public override bool IsValid( Dictionary<string, Operation> path, RuleContext context, out object[] formatParameters)
+        {
+            List<string> notAllowedProperties = new List<string>();
+            foreach (var operation in path.Keys)
+            {
+                if ((operation.ToLower().Equals("get") ||
+                    operation.ToLower().Equals("put") ||
+                    operation.ToLower().Equals("patch")) && path[operation].Parameters != null)
+                {
+                    foreach (var param in path[operation].Parameters)
+                    {
+                        if (param.In == ParameterLocation.Body)
+                        {
+                            if (param.Schema.Reference != null)
+                            {
+                                string defName = param.Schema.Reference.Split('/').Last();
```

Comment:
nit: you can use https://github.com/Azure/autorest/blob/e2af872f328678a6d6bddb44fc15cb5efc5b7b0c/src/modeler/AutoRest.Swagger/Extensions.cs#L54 instead